src/vstest.console/Processors/ListExtensionsArgumentProcessor.cs

```diff
@@ -0,0 +1,243 @@
+        {
+        }
+
+        public ArgumentProcessorResult Execute()
+        {
+            ConsoleOutput.Instance.WriteLine(CommandLineResources.AvailableLoggersHeaderMessage, OutputLevel.Information);
+            var testPlatform = TestPlatformFactory.GetTestPlatform();
+            var extensionManager = TestLoggerExtensionManager.Create(new NullMessageLogger());
+            foreach (var extension in extensionManager.TestExtensions)
+            {
+                ConsoleOutput.Instance.WriteLine(extension.Value.GetType().FullName, OutputLevel.Information);
+                ConsoleOutput.Instance.WriteLine(string.Format(CultureInfo.CurrentCulture, CommandLineResources.AvailableExtensionsMetadataFormat, "Uri", extension.Metadata.ExtensionUri), OutputLevel.Information);
+                ConsoleOutput.Instance.WriteLine(string.Format(CultureInfo.CurrentCulture, CommandLineResources.AvailableExtensionsMetadataFormat, "FriendlyName", string.Join(", ", extension.Metadata.FriendlyName)), OutputLevel.Information);
+            }
+
+            return ArgumentProcessorResult.Success;
+        }
+
+        private class NullMessageLogger : IMessageLogger
+        {
+            public void SendMessage(TestMessageLevel testMessageLevel, string message)
+            {
+            }
+        }
+    }
+    #endregion
+
+    #region List settings providers
+    /// <summary>
+    /// Argument Executor for the "/ListSettingsProviders" command line argument.
+    /// </summary>
+    internal class ListSettingsProvidersArgumentProcessor : ListExtensionsArgumentProcessor
+    {
+        private const string CommandName = "/ListSettingsProviders";
+
+        public ListSettingsProvidersArgumentProcessor()
+            : base(() => new ListSettingsProvidersArgumentExecutor(), () => new ListExtensionsArgumentProcessorCapabilities(CommandName))
+        {
+        }
+    }
+
+    internal class ListSettingsProvidersArgumentExecutor : IArgumentExecutor
+    {
+        public void Initialize(string argument)
+        {
+        }
+
+        public ArgumentProcessorResult Execute()
+        {
+            ConsoleOutput.Instance.WriteLine(CommandLineResources.AvailableSettingsProvidersHeaderMessage, OutputLevel.Information);
+            var testPlatform = TestPlatformFactory.GetTestPlatform();
+            var extensionManager = SettingsProviderExtensionManager.Create();
+            foreach (var extension in extensionManager.SettingsProvidersMap.Values)
+            {
+                ConsoleOutput.Instance.WriteLine(extension.Value.GetType().FullName, OutputLevel.Information);
+                ConsoleOutput.Instance.WriteLine(string.Format(CultureInfo.CurrentCulture, CommandLineResources.AvailableExtensionsMetadataFormat, "SettingName", extension.Metadata.SettingsName), OutputLevel.Information);
+            }
+
+            return ArgumentProcessorResult.Success;
+        }
+
+        private class NullMessageLogger : IMessageLogger
+        {
```

Comment:
Do we require NullMessageLogger here?

Reply:
It is required by this argument processor.

Reply:
@chakradhar2 thanks, I've removed the extraneous code.